File: src/main/java/hytale/academy/template/TemplatePlugin.java
package hytale.academy.template;

import com.hypixel.hytale.server.core.plugin.JavaPlugin;
import com.hypixel.hytale.server.core.plugin.JavaPluginInit;
import hytale.academy.template.command.PingPlayerCommand;
import org.checkerframework.checker.nullness.compatqual.NonNullDecl;

public class TemplatePlugin extends JavaPlugin {

    public TemplatePlugin(@NonNullDecl JavaPluginInit init) {
        super(init);
    }

    @Override
    protected void setup() {
        getCommandRegistry().registerCommand(new PingPlayerCommand());
    }
}

File: src/main/kotlin/org/hytaleacademy/template/TemplatePlugin.kt
package org.hytaleacademy.template

import com.hypixel.hytale.server.core.event.events.player.PlayerReadyEvent
import com.hypixel.hytale.server.core.plugin.JavaPlugin
import com.hypixel.hytale.server.core.plugin.JavaPluginInit
import org.hytaleacademy.template.command.PingPlayerCommand
import org.hytaleacademy.template.listener.PlayerReadyListener

class TemplatePlugin(init: JavaPluginInit) : JavaPlugin(init) {

    override fun setup() {
        commandRegistry.registerCommand(PingPlayerCommand())
        eventRegistry.registerGlobal(PlayerReadyEvent::class.java, PlayerReadyListener::onPlayerJoin)
    }
}